.github/workflows/prototype-tests.yml

```diff
@@ -40,13 +40,18 @@ jobs:
         run: pip install --progress-bar=off pytest pytest-mock
 
       - name: Run prototype datasets tests
+        id: datasets
         shell: bash
         run: pytest --durations=20 test/test_prototype_datasets*.py
 
       - name: Run prototype transforms tests
+        id: transforms
+        if: ${{ success() || steps.datasets.conclusion == 'failure' }}
         shell: bash
         run: pytest --durations=20 test/test_prototype_transforms*.py
 
       - name: Run prototype models tests
+        if: ${{ success() || steps.datasets.conclusion == 'failure' || steps.transforms.conclusion == 'failure' }}
```

Comment:
@pmeier If I understand correctly, since we have:
```
if: ${{ success() || steps.datasets.conclusion == 'failure' }}
```
on the transforms test, therefore it will not be skipped and will have state of either `success` or `failure`.

In this case, I think we dont need `steps.datasets.conclusion == 'failure'` conditions here since it is kinda redundant. I might be wrong on this though, could you confirm this @pmeier ?

Reply:
> therefore it will not be skipped and will have state of either `success` or `failure`.

`failure` is not concrete enough. It will not be skipped in case of failure of the datasets tests. If the failure happens in any other step, e.g. `torchvision` installation, the step will be skipped.

> In this case, I think we dont need `steps.datasets.conclusion == 'failure'` conditions here since it is kinda redundant.

They are not redundant. If we leave them out, we are left with `if: success()` which is the default behavior if you don't define `if:` at all. Meaning, the step will only be run if *all* previous steps have succeeded. 

We need this step to run if *all setup steps* (everything before the first test step) have succeeded, regardless of the state of the datasets tests. IMO `success() || steps.datasets.conclusion == 'failure'` is shortest expression that gets us this behavior, but if you can find a better one, I'm all ears.